app/Http/Controllers/CategoryController.php

```diff
@@ -28,21 +29,25 @@ class CategoryController extends Controller
     public function index($name,Request $request)
     {		
 	  
-        $categories = Category::all()->toArray();
-        $category = Category::where('name',$name)->get()->toArray();
+        $categories = Category::all();
+        $category = Category::where('name',$name)->get();
 		
 		if(count($category)>0){
-			$products = Product::leftJoin('category_product', 'product_id', '=', 'products.id')->leftJoin('media', 'model_id', '=', 'products.id')->where('category_id',$category[0]['id'])->select('products.id as product_id','products.*','file_name','media.id as media_id')->get()->toArray();
 			
+			$products = CategoryProduct::with(['product.media'])->where('category_id',$category[0]['id'])->get();
```

Comment:
If you need only one record in $category then you can use first() instead of get() function. then you don't need to pass 0 index here.